### PR TITLE
[terminal] Fixed wrong checks

### LINKED ISSUE #5431: [terminal] link matcher doesn't work with rust / go cli output
Rust and go use the following scheme:

relative/path:start:end:

which is not recognized.

Example for go:
```
# github.com/demo-apps/go-gin-app
./models.article.go:16:11: undefined: leList
```

Example for Rust:
```
error[E0061]: this function takes 1 parameter but 2 parameters were supplied
  --> src/app.rs:45:10
   |
45 |         .arg(
   |          ^^^ expected 1 parameter

error[E0599]: no function or associated item named `with` found for type `clap::Arg<'_, '_>` in the current scope
  --> src/app.rs:46:18
   |
46 |             Arg::with _name("case-sensitive")
   |                  ^^^^ function or associated item not found in `clap::Arg<'_, '_>`

```

### PR DESCRIPTION
**Pull request by svenefftinge**
fixes #5431

### CODE REVIEW

**Comment by svenefftinge** [comment]
The code was previously comparing widget ids (strings) with terminal ids (number). Just calling `toString()` on that doesn't work of course. That's why we use types.

File: packages/process/src/node/process.ts [hunk 1]
```diff
@@ -195,10 +195,6 @@ export abstract class Process {
                 });
             });
         } else if (!isWindows) {
-            return new Promise<string>(resolve => {
```
Comment (akosyakov) [inline packages/process/src/node/process.ts]: good catch! :)

**Comment by vince-fugnitto** [comment]
@svenefftinge sorry about that, I believe I was confused by which `id` was actually being used.
We will need to fix the PR https://github.com/theia-ide/theia/pull/5365 again which isn't a big deal :)

File: packages/process/src/node/process.ts (hunk 1 above)
Comment (svenefftinge) [reply packages/process/src/node/process.ts]: Yes, thank you :)

**Pull request merged**